components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/dao/OAuthDAO.java

```diff
@@ -816,151 +995,249 @@ public void deleteOAuthToken(TempOAuthToken deleteAuthToken, KeyRotationConfig k
-        preparedStatement.setString(21, oAuthToken.getTokenBindingRef());
-        preparedStatement.setString(22, oAuthToken.getAccessToken());
-        preparedStatement.setString(23, oAuthToken.getRefreshToken());
-        preparedStatement.setInt(24, Integer.parseInt(oAuthToken.getConsumerKeyId()));
-        preparedStatement.setString(25, oAuthToken.getAuthzUser());
-        preparedStatement.setInt(26, Integer.parseInt(oAuthToken.getTenantId()));
-        preparedStatement.setString(27, oAuthToken.getUserDomain());
-        preparedStatement.setString(28, oAuthToken.getUserType());
-        preparedStatement.setString(29, oAuthToken.getGrantType());
-        preparedStatement.setTimestamp(30, Timestamp.valueOf(oAuthToken.getTimeCreated()));
-        preparedStatement.setTimestamp(31,
-                Timestamp.valueOf(oAuthToken.getRefreshTokenTimeCreated()));
-        preparedStatement.setInt(32, Integer.parseInt(oAuthToken.getValidityPeriod()));
-        preparedStatement.setInt(33,
-                Integer.parseInt(oAuthToken.getRefreshTokenValidityPeriod()));
-        preparedStatement.setString(34, oAuthToken.getTokenScopeHash());
-        preparedStatement.setString(35, oAuthToken.getTokenState());
-        preparedStatement.setString(36, oAuthToken.getTokenStateId());
-        preparedStatement.setString(37, oAuthToken.getSubjectIdentifier());
-        preparedStatement.setString(38, oAuthToken.getAccessTokenHash());
-        preparedStatement.setString(39, oAuthToken.getRefreshTokenHash());
-        preparedStatement.setInt(40, Integer.parseInt(oAuthToken.getIdpId()));
-        preparedStatement.setString(41, oAuthToken.getTokenBindingRef());
+        List<TempOAuthScope> tempOAuthScopeList = new ArrayList<>();
+        try (Connection connection = DriverManager
+                .getConnection(keyRotationConfig.getOldIdnDBUrl(), keyRotationConfig.getOldIdnUsername(),
+                        keyRotationConfig.getOldIdnPassword())) {
+            connection.setAutoCommit(false);
+            try (PreparedStatement preparedStatement = connection.prepareStatement(DBConstants.GET_TEMP_OAUTH_SCOPE)) {
+                preparedStatement.setInt(1, syncId);
+                ResultSet resultSet = preparedStatement.executeQuery();
+                connection.commit();
+                while (resultSet.next()) {
+                    tempOAuthScopeList.add(new TempOAuthScope(resultSet.getString(KeyRotationConstants.TOKEN_ID),
+                            resultSet.getString(KeyRotationConstants.TOKEN_SCOPE),
+                            resultSet.getString(KeyRotationConstants.TENANT_ID),
+                            resultSet.getInt(KeyRotationConstants.AVAILABILITY),
+                            resultSet.getInt(KeyRotationConstants.SYNC_ID),
+                            resultSet.getInt(KeyRotationConstants.SYNCED)));
+                }
+            } catch (SQLException e) {
+                connection.rollback();
```

Comment:
Do we need to rollback the connection here. Since this is just a DB read

Reply:
Yes, we don't. Addressed with https://github.com/wso2/identity-tools/pull/22/commits/fc75ada5bb072ae87d82ca1e21429d0e1d3a9ef9